File: src/main/java/com/icloud/modules/unimall/entity/UnimallSpu.java
package com.icloud.modules.unimall.entity;

import com.baomidou.mybatisplus.annotation.TableName;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableField;
import lombok.Data;

import java.io.Serializable;
import java.util.Date;

/**
 * 
 * 
 * @author zdh
 * @email yyyyyy@cm.com
 * @date 2020-05-16 10:38:18
 */
@Data
@TableName("t_unimall_spu")
public class UnimallSpu implements Serializable {
	private static final long serialVersionUID = 1L;

   	   /*  */
       @TableId(value="id", type= IdType.AUTO)
       private Long id;
   	   	   /* 原价 */
       @TableField("original_price")
       private Integer originalPrice;
   	   	   /* 现价 */
       @TableField("price")
       private Integer price;
   	   	   /* vip价 */
       @TableField("vip_price")
       private Integer vipPrice;
   	   	   /* 商品名称 */
       @TableField("title")
       private String title;
   	   	   /* 销量 */
       @TableField("sales")
       private Integer sales;
   	   	   /* 商品图片 */
       @TableField("img")
       private String img;
   	   	   /* 商品详情 */
       @TableField("detail")
       private String detail;
   	   	   /* 商品描述 */
       @TableField("description")
       private String description;
   	   	   /* 分类id */
       @TableField("category_id")
       private Long categoryId;
   	   	   /* 运费模板id */
       @TableField("freight_template_id")
       private Long freightTemplateId;
   	   	   /* 计量单位 */
       @TableField("unit")
       private String unit;
   	   	   /* 0下架 1上架 */
       @TableField("status")
       private Integer status;
   	   	   /*  */
       @TableField("gmt_update")
       private Date gmtUpdate;
   	   	   /*  */
       @TableField("gmt_create")
       private Date gmtCreate;
   	   	   /* 商户id */
       @TableField("supplier_id")
       private Long supplierId;

    /*关联supplier*/
    @TableField(exist = false)
    private UnimallSupplier supplier;
   	
}

File: src/main/resources/statics/js/modules/unimall/unimallspu.js
$(function () {
    $("#jqGrid").jqGrid({
        url: baseURL + 'unimall/unimallspu/list',
        datatype: "json",
        colModel: [			
			{ label: 'id', name: 'id', index: 'id', width: 50, key: true },
			{ label: '原价', name: 'originalPrice', index: 'original_price', width: 80 }, 			
			{ label: '现价', name: 'price', index: 'price', width: 80 }, 			
			{ label: 'vip价', name: 'vipPrice', index: 'vip_price', width: 80 }, 			
			{ label: '商品名称', name: 'title', index: 'title', width: 80 }, 			
			{ label: '销量', name: 'sales', index: 'sales', width: 80 }, 			
			{ label: '商品图片', name: 'img', index: 'img', width: 80 }, 			
			{ label: '商品详情', name: 'detail', index: 'detail', width: 80 }, 			
			{ label: '商品描述', name: 'description', index: 'description', width: 80 }, 			
			{ label: '分类id', name: 'categoryId', index: 'category_id', width: 80 }, 			
			{ label: '运费模板id', name: 'freightTemplateId', index: 'freight_template_id', width: 80 }, 			
			{ label: '计量单位', name: 'unit', index: 'unit', width: 80 }, 			
			{ label: '0下架 1上架', name: 'status', index: 'status', width: 80 }, 			
			{ label: '', name: 'gmtUpdate', index: 'gmt_update', width: 80 }, 			
			{ label: '', name: 'gmtCreate', index: 'gmt_create', width: 80 }, 			
			{ label: '商户id', name: 'supplierId', index: 'supplier_id', width: 80 }			
        ],
		viewrecords: true,
        height: 385,
        rowNum: 10,
		rowList : [10,30,50],
        rownumbers: true, 
        rownumWidth: 25, 
        autowidth:true,
        multiselect: true,
        pager: "#jqGridPager",
        jsonReader : {
            root: "page.list",
            page: "page.currPage",
            total: "page.totalPage",
            records: "page.totalCount"
        },
        prmNames : {
            page:"page", 
            rows:"limit", 
            order: "order"
        },
        gridComplete:function(){
        	//隐藏grid底部滚动条
        	$("#jqGrid").closest(".ui-jqgrid-bdiv").css({ "overflow-x" : "hidden" }); 
        }
    });
});

var vm = new Vue({
	el:'#icloudapp',
	data:{
		showList: true,
		title: null,
		unimallSpu: {}
	},
	methods: {
		query: function () {
			vm.reload();
		},
		add: function(){
			vm.showList = false;
			vm.title = "新增";
			vm.unimallSpu = {};
		},
		update: function (event) {
			var id = getSelectedRow();
			if(id == null){
				return ;
			}
			vm.showList = false;
            vm.title = "修改";
            
            vm.getInfo(id)
		},
		saveOrUpdate: function (event) {
		    $('#btnSaveOrUpdate').button('loading').delay(1000).queue(function() {
                var url = vm.unimallSpu.id == null ? "unimall/unimallspu/save" : "unimall/unimallspu/update";
                $.ajax({
                    type: "POST",
                    url: baseURL + url,
                    contentType: "application/json",
                    data: JSON.stringify(vm.unimallSpu),
                    success: function(r){
                        if(r.code === 0){
                             layer.msg("操作成功", {icon: 1});
                             vm.reload();
                             $('#btnSaveOrUpdate').button('reset');
                             $('#btnSaveOrUpdate').dequeue();
                        }else{
                            layer.alert(r.msg);
                            $('#btnSaveOrUpdate').button('reset');
                            $('#btnSaveOrUpdate').dequeue();
                        }
                    }
                });
			});
		},
		del: function (event) {
			var ids = getSelectedRows();
			if(ids == null){
				return ;
			}
			var lock = false;
            layer.confirm('确定要删除选中的记录？', {
                btn: ['确定','取消'] //按钮
            }, function(){
               if(!lock) {
                    lock = true;
		            $.ajax({
                        type: "POST",
                        url: baseURL + "unimall/unimallspu/delete",
                        contentType: "application/json",
                        data: JSON.stringify(ids),
                        success: function(r){
                            if(r.code == 0){
                                layer.msg("操作成功", {icon: 1});
                                $("#jqGrid").trigger("reloadGrid");
                            }else{
                                layer.alert(r.msg);
                            }
                        }
				    });
			    }
             }, function(){
             });
		},
		getInfo: function(id){
			$.get(baseURL + "unimall/unimallspu/info/"+id, function(r){
                vm.unimallSpu = r.unimallSpu;
            });
		},
		reload: function (event) {
			vm.showList = true;
			var page = $("#jqGrid").jqGrid('getGridParam','page');
			$("#jqGrid").jqGrid('setGridParam',{ 
                page:page
            }).trigger("reloadGrid");
		}
	}
});